File: python3/LeetCode/findLongestWord.py
from typing import List


class Solution:
    def findLongestWord(self, s: str, dictionary: List[str]) -> str:
        dictionary.sort(key=lambda x: (-len(x), x))
        for dic in dictionary:
            left=0
            right=0
            while left<len(dic) and right<len(s):
                if dic[left] == s[right]:
                    left+=1
                    right+=1
                else:
                    right+=1

            if left == len(dic):
                return dic

        return ""

File: go/LeetCode/findLongestWord.go
package main

import "sort"

func main() {
	
}

func findLongestWord(s string, dictionary []string) string {
	sort.Slice(dictionary, func(i, j int) bool {
		a,b :=dictionary[i], dictionary[j]
		return len(a)>len(b) || len(a) == len(b) && a<b
	})

	for _,v := range dictionary{
		left:=0
		right:=0
		for left<len(v) && right<len(s) {
			if v[left] == s[right] {
				left++
				right++
			}else {
				right++
			}
		}
		if left == len(v) {
			return v
		}
	}
	return ""
}
